### PR TITLE
WINDUP-997: Reworked validation and removed mock tests in exchange fo…

### PR DESCRIPTION
**Pull request by jsight**
…r real-world tests

### CODE REVIEW

**Comment by mareknovotny** [comment]
retest

**Comment by mareknovotny** [comment]
it seems there are still 3 failed tests in nested condition test and 2 in integration tests

**Comment by OndraZizka** [comment]
http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1685/

File: rules-xml/tests/src/test/resources/xsd-validation/xsd-url-multiple-schemas.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<beans xmlns="http://www.springframework.org/schema/beans"
```
Comment (mareknovotny) [inline rules-xml/tests/src/test/resources/xsd-validation/xsd-url-multiple-schemas.xml]: this new file affects test https://github.com/windup/windup/blob/master/rules-xml/tests/src/test/java/org/jboss/windup/rules/xml/XMLFileNestedConditionTest.java#L127

**Comment by mareknovotny** [comment]
it affects effort *2210* in https://github.com/windup/windup/blob/master/tests/src/test/java/org/jboss/windup/tests/application/WindupArchitectureMediumBinaryModeTest.java#L163 as where are hints with"XML File is not valid" now and `WindupArchitectureDuplicateTest#testRunWindupDuplicateEAR()` where are not counted with Potential issues for XML invalid file?

**Comment by jsight** [comment]
Yes, and these were actual bugs as well. :) Reworked to fix them... hopefully this is more stable now.

File: rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/ValidateXmlFilesRuleProvider.java [hunk 2]
```diff
@@ -85,139 +72,198 @@ public Configuration getConfiguration(GraphContext context)
...
+                        parseExceptions.add(e);
+                    }
+
+                    @Override
+                    public void fatalError(SAXParseException e) throws SAXException
+                    {
+                        parseExceptions.add(e);
+                    }
+                };
+
```
Comment (OndraZizka) [inline rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/ValidateXmlFilesRuleProvider.java]: Maybe this class would deserve moving to a separate file? To make the validator RP a bit more readable...

File: rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/EnhancedEntityResolver2.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+        int status = httpConnection.getResponseCode();
+        if ((status != HttpURLConnection.HTTP_OK) &&
+                    (status == HttpURLConnection.HTTP_MOVED_TEMP
+                                || status == HttpURLConnection.HTTP_MOVED_PERM
+                                || status == HttpURLConnection.HTTP_SEE_OTHER))
+        {
+            String newUrl = httpConnection.getHeaderField("Location");
+            httpConnection = (HttpURLConnection) new URL(newUrl).openConnection();
+        }
```
Comment (OndraZizka) [inline rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/EnhancedEntityResolver2.java]: What if it's behind seveal redirects? ;-)

Comment (mareknovotny) [inline rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/EnhancedEntityResolver2.java]: @OndraZizka URL can change or move at least for older things.

File: rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/ValidateXmlFilesRuleProvider.java (hunk 2 above)
Comment (jsight) [inline rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/ValidateXmlFilesRuleProvider.java]: Yeah, this class is starting to get more complicated. Will revisit if we hit more bugs.

File: rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/EnhancedEntityResolver2.java (hunk 3 above)
Comment (jsight) [inline rules-xml/impl/src/main/java/org/jboss/windup/rules/apps/xml/xml/EnhancedEntityResolver2.java]: @OndraZizka It wouldn't surprise me if it has to be a loop at some point. I wanted to just set "setFollowRedirects", but that isn't sufficient for some cases.

**Pull request merged**